hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/SnapshotCache.java

```diff
@@ -296,105 +262,34 @@ private synchronized void cleanup() {
    * TODO: [SNAPSHOT] Add new ozone debug CLI command to trigger this directly.
    */
   private void cleanupInternal() {
-    long numEntriesToEvict = (long) dbMap.size() - cacheSizeLimit;
-    while (numEntriesToEvict > 0L && pendingEvictionList.size() > 0) {
-      // Get the first instance in the clean up list
-      ReferenceCounted<IOmMetadataReader, SnapshotCache> rcOmSnapshot =
-          pendingEvictionList.iterator().next();
+    Iterator<Map.Entry<String, ReferenceCounted<IOmMetadataReader, SnapshotCache>>> iterator =
+        dbMap.entrySet().iterator();
+    while (iterator.hasNext()) {
```

Comment:
I don't see a stopping condition other than reaching the end of the `Map` entry set here. So it closes ALL snapshots with zero RC.

It works but it might be inefficient.

But adding back `numEntriesToEvict` would also cause fairness issue (plus it also won't overcome the inefficiency issue if the LAST element in the `Map` has zero RC).

Is there any plan to improve this later on?

Reply:
We can have follow up task for it.

Reply:
Follow up task to optimize the clean up HDDS-10156